command/do_transition_test.go

```diff
@@ -0,0 +1,98 @@
+package command
+
+import (
+	"encoding/json"
+	"github.com/ExpediaGroup/flyte-jira/client"
+	"github.com/HotelsDotCom/flyte-client/flyte"
+	"io/ioutil"
+	"net/http"
+	"reflect"
+	"testing"
+)
+
+type (
+	mockRequestBody struct {
+		TransitionId string `json:"id"`
+	}
+
+	DoTransitionRequest struct {
+		Transition mockRequestBody `json:"transition"`
+	}
+)
+
+func createMockSendRequest(issueId, transitionId string) func(request *http.Request) (int, error) {
+	return func(request *http.Request) (int, error) {
+		b, err := ioutil.ReadAll(request.Body)
+		if err != nil {
+			return 400, err
+		}
+
+		body := &DoTransitionRequest{}
+
+		if err := json.Unmarshal(b, &body); err != nil {
+			return 400, err
+		}
+
+		if body.Transition.TransitionId != transitionId {
+			return 500, err
+		}
+
+		if issueId != "DEVEX-123" {
+			return 404, err
+		}
+
+		return 204, nil
+	}
+}
+func TestDoTransitionAsExpected(t *testing.T) {
+	client.SendRequestWithoutResp = createMockSendRequest("DEVEX-123", "881")
+	input := []byte(`{"issueId":"DEVEX-123","transitionId":"881"}`)
+	actual := doTransitionHandler(input)
+	exp := flyte.Event{
+		EventDef: doTransitionEventDef,
+		Payload: doTransitionRequest{
+			IssueId:      "DEVEX-123",
+			TransitionId: "881",
+		},
+	}
+
+	if !reflect.DeepEqual(actual, exp) {
```

Comment:
Will be good to use `testify` package for assertions

`"github.com/stretchr/testify/assert"` then rather than write this awkward syntax you should be able to use
`assert.Equal(t, want, got)`

Reply:
done